pkg/node/chain.go

```diff
@@ -0,0 +1,249 @@
+	ctx context.Context,
+	log logging.Logger,
+	stateStore storage.StateStorer,
+	endpoint string,
+	signer crypto.Signer,
+) (
+	backend *ethclient.Client,
+	overlayEthAddress common.Address,
+	chainID int64,
+	transactionService transaction.Service,
+	err error,
+) {
+	backend, err = ethclient.Dial(endpoint)
+	if err != nil {
+		err = fmt.Errorf("failed to connect to the Ethereum backend at %s: %w", endpoint, err)
+		return
+	}
+
+	transactionService, err = transaction.NewService(log, backend, signer, stateStore)
+	if err != nil {
+		err = fmt.Errorf("failed to init transaction service: %w", err)
+		return
+	}
+
+	overlayEthAddress, err = signer.EthereumAddress()
+	if err != nil {
+		err = fmt.Errorf("failed to determine overlay Ethereum address: %w", err)
+		return
+	}
+
+	cID, err := backend.ChainID(ctx)
+	if err != nil {
+		log.Infof(
+			"could not connect to backend at %s. With swap enabled on the"+
+				" network a working blockchain node (for goerli network in"+
+				" production) is required. Check your node or specify another node"+
+				" endpont using --swap-endpoint",
+			endpoint,
+		)
+		err = fmt.Errorf("could not get chain ID from endpoint %s: %w", endpoint, err)
+		return
+	}
+	chainID = cID.Int64()
+
+	return
+}
+
+// InitChequebookFactory will initialize the chequebook factory with the given
+// chain backend.
+func InitChequebookFactory(
+	log logging.Logger,
+	backend *ethclient.Client,
+	chainID int64,
+	transactionService transaction.Service,
+	factoryAddress string,
+) (
+	chequebookFactory chequebook.Factory,
+	err error,
+) {
+	var addr common.Address
+	if factoryAddress == "" {
+		var found bool
+		addr, found = chequebook.DiscoverFactoryAddress(chainID)
+		if !found {
+			return nil, errors.New("no known factory address for this network")
+		}
+		log.Infof("using default factory address for chain id %d: %x", chainID, addr)
+	} else if !common.IsHexAddress(factoryAddress) {
+		return nil, errors.New("malformed factory address")
+	} else {
+		addr = common.HexToAddress(factoryAddress)
+		log.Infof("using custom factory address: %x", factoryAddress)
+	}
+
+	chequebookFactory, err = chequebook.NewFactory(
+		backend,
+		transactionService,
+		addr,
+		chequebook.NewSimpleSwapFactoryBindingFunc,
+	)
+	if err != nil {
+		err = fmt.Errorf("failed to init chequebook factory: %w", err)
+	}
+
+	return
+}
+
+// InitChequebookService will initialize the chequebook service with the given
+// chequebook factory and chain backend.
+func InitChequebookService(
+	ctx context.Context,
+	log logging.Logger,
+	stateStore storage.StateStorer,
+	signer crypto.Signer,
+	chainID int64,
+	backend *ethclient.Client,
+	overlayEthAddress common.Address,
+	transactionService transaction.Service,
+	chequebookFactory chequebook.Factory,
+	initialDeposit string,
+) (
+	chequebookService chequebook.Service,
+	err error,
+) {
+	chequeSigner := chequebook.NewChequeSigner(signer, chainID)
+
+	isSynced, err := transaction.IsSynced(ctx, backend, maxDelay)
```

Comment:
does this not belong better to `InitChain`?

Reply:
I would agree, I just wasn't sure if there is a case where we want to initialize the chain backend without explicitly syncing it. I will move it there.